platform/linuxbsd/joypad_linux.cpp

```diff
@@ -536,7 +655,9 @@ void JoypadLinux::process_joypads() {
 
 			switch (joypad_event.type) {
 				case EV_KEY:
-					input->joy_button(i, (JoyButton)joypad.key_map[joypad_event.code], joypad_event.value);
+					if (joypad.key_map[joypad_event.code] >= 0) {
```

Comment:
There may be a situation where one button has two events, or two buttons are mapped to one key. To prevent this when **auto map** is enabled, some existing keys will be mapped to the negative range.

Reply:
Yes, I know.  
And you're right. While `ERR_FAIL_INDEX` would catch the same case and abort, it would also print error messages…